tests/Database/DatabaseEloquentIntegrationTest.php

```diff
@@ -2101,6 +2107,21 @@ public function testMorphPivotsCanBeRefreshed()
         $this->assertSame('primary', $pivot->taxonomy);
     }
 
+    public function testFillDateWithCustomCast()
+    {
+        $user = EloquentDateCastingUser::create([
+            'date_of_birth' => now()
+        ]);
+
+        $user->refresh();
+
+        $user->fill(['date_of_birth' => '20-10-2000']);
+
+        $user->save();
+
```

Comment:
What is the value in the database after this update (`$user->refresh();`)? It might be SQlite just parses the d-m-Y string as a different value and ignores any format error.

Reply:
I only added this to get as close to your issue as possible (since I figured the `$user` on which you were calling `fill()->save()` probably was created before the lifecycle of the request).

It was still failing without it, though.

Reply:
> Is the test failing or was that a typo?

Yes that was a typo. It was still passing.